tools/azure-pipelines/e2e-template.yml

```diff
@@ -98,6 +98,19 @@ jobs:
 
         echo "Free up disk space"
         ./tools/azure-pipelines/free_disk_space.sh
+        
+        # the APT mirrors access is based on a proposal from https://github.com/actions/runner-images/issues/7048#issuecomment-1419426054
+        echo "Configure APT mirrors"
+        mirror_file_path="/etc/apt/mirrors.txt"
+        default_ubuntu_mirror_url="http://azure.archive.ubuntu.com/ubuntu/"
+        
+        # add Azure's Ubuntu mirror as a top-priority source
+        echo -e "${default_ubuntu_mirror_url}\tpriority:1" | sudo tee ${mirror_file_path}
+        
+        # use other mirrors as a fallback option
+        curl http://mirrors.ubuntu.com/mirrors.txt | sudo tee --append ${mirror_file_path}
```

Comment:
The only thing this could introduce is option for a supply chain attack, but we already have others sources where we do something similar and I would classify the risk that Ubuntu gets compromised the same as we do for the other sources (Maven/Github).

Reply:
Interesting point. Just to understand that: It makes us vulnerable because we're relying on `mirrors.ubuntu.com/mirrors.txt`? What about specifying a fixed list of mirrors ourselves instead of relying on an external service?

Reply:
I guess, that's reasonable. I will update the branch accordingly. Thanks for bringing it up :-)

Reply:
Any objections against that list?